### ISSUE TITLE
would you mind sharing one instruction in detail? 

### ISSUE THREAD

**Issue by Ai-is-light** [FORTH-ModelBasedTracker/MocapNET#2]
--------------------------------------------------------------------
after building,  I would like to testing the repo, while I have to use the centOS service or the docker environment without camera. So, I tried to process the video by your repo and save the processed output at local path. But I can not find one detailed instruction about this excellent work?
I got the following error:
root@277aeac2e1bb:/workspace/pcode_tme/pose_dancing/MocapNET# ./WebcamBIN --from ./video_test/sense_test.mp4
Trying to open ./video_test/sense_test.mp4
QXcbConnection: Could not connect to display
Aborted
root@277aeac2e1bb:/workspace/pcode_tme/pose_dancing/MocapNET# vim ~/.bashrc
root@277aeac2e1bb:/workspace/pcode_tme/pose_dancing/MocapNET# source ~/.bashrc
root@277aeac2e1bb:/workspace/pcode_tme/pose_dancing/MocapNET# ./WebcamBIN --from ./video_test/sense_test.mp4
Trying to open ./video_test/sense_test.mp4
QFontDatabase: Cannot find font directory /usr/lib/x86_64-linux-gnu/fonts - is Qt installed correctly?
QFontDatabase: Cannot find font directory /usr/lib/x86_64-linux-gnu/fonts - is Qt installed correctly?
QFontDatabase: Cannot find font directory /usr/lib/x86_64-linux-gnu/fonts - is Qt installed correctly?
QFontDatabase: Cannot find font directory /usr/lib/x86_64-linux-gnu/fonts - is Qt installed correctly?
This plugin does not support propagateSizeHints()
This plugin does not support propagateSizeHints()
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /root/opencv/modules/highgui/src/window.cpp, line 339
terminate called after throwing an instance of 'cv::Exception'
  what():  /root/opencv/modules/highgui/src/window.cpp:339: error: (-215) size.width>0 && size.height>0 in function imshow
Aborted
------------------------------------------------------------------
And, I guess I built it successfully under the mac's docker environment
MocapNET: Input was given precompressed
Direction is : 38.503 Front
Sample 195/200 - 10.0350ms - mae 1.4426
MocapNET: Input was given precompressed
Direction is : 34.6637 Front
Sample 196/200 - 10.0750ms - mae 1.4900
MocapNET: Input was given precompressed
Direction is : 46.3801 Front
Sample 197/200 - 9.0950ms - mae 1.4359
MocapNET: Input was given precompressed
Direction is : 70.9141 Front
Sample 198/200 - 9.0400ms - mae 1.2482
MocapNET: Input was given precompressed
Direction is : 70.4886 Front
Sample 199/200 - 12.7740ms - mae 1.2491
model name	: Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz

Total 9986.97ms for 1000 samples - Average 9.98697ms - 100.13 fps
root@277aeac2e1bb:/workspace/pcode_tme/pose_dancing/MocapNET# ./MocapNETBenchmark --cpu^C

**Comment by AmmarkoV** [comment]
Thank you for your kind words..!
This appears to be an OpenCV related error..! Since you where able to run the benchmark it means that the tensorflow and neural network are working correctly.
Today I will try to make the code handle the visualization windows  to make them to be able to be switched off.
The error appears to be OpenCV related.. And I think it must have to do with the headless docker container. ( as the QXcbConnection: Could not connect to display ) also suggests.
and the 
 Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /root/opencv/modules/highgui/src/window.cpp, line 339

The Demo was supposed to be working with a screen and output a visualization window but I will make this optional and when this is ready you can git pull recompile and it will work..!

**Comment by AmmarkoV** [comment]
I added a flag : --novisualization that suppresses visualization windows

You should git pull , recompile and then try ./WebcamBIN --from ./video_test/sense_test.mp4 --novisualization and see if it will now work on the docker container..

https://github.com/FORTH-ModelBasedTracker/MocapNET/commit/d5c7471cc2e083fc1a3e21f3f7cb07c85d410836

**Comment by Ai-is-light** [comment]
now, I can get the image imshow at docker environment.
and, --novisualizaition man not work?
I run root@7953ba7131a3:/pose_dancing/we-try-git-newest/MocapNET# ./WebcamBIN --from video_test/sense_test.mp4  --novisualization
I got as following:
Trying to open video_test/sense_test.mp4
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp, line 304
terminate called after throwing an instance of 'cv::Exception'
  what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp:304: error: (-215) size.width>0 && size.height>0 in function imshow.
Aborted

It can work for about 10 seconds with the image of video_test/sense_test.mp4. However, nothing have been processed. I'm a little bit confuesed with it. Anyway, with the old version of this repo, I got the result out keypoints and others, but the old version can only work with the first some frames, I will be Aborted, which may be caused by so many windows with in dokcer.

And, there the latest results. Thanks for your updating and replies.
![Uploading image.png…]()


> I added a flag : --novisualization that suppresses visualization windows
> 
> You should git pull , recompile and then try ./WebcamBIN --from ./video_test/sense_test.mp4 --novisualization and see if it will now work on the docker container..
> 
> [d5c7471](https://github.com/FORTH-ModelBasedTracker/MocapNET/commit/d5c7471cc2e083fc1a3e21f3f7cb07c85d410836)

now, I can get the image imshow at docker environment.
and, --novisualizaition man not work?
I run root@7953ba7131a3:/pose_dancing/we-try-git-newest/MocapNET# ./WebcamBIN --from video_test/sense_test.mp4  --novisualization
I got as following:
Trying to open video_test/sense_test.mp4
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp, line 304
terminate called after throwing an instance of 'cv::Exception'
  what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp:304: error: (-215) size.width>0 && size.height>0 in function imshow.
Aborted

It can work for about 10 seconds with the image of video_test/sense_test.mp4. However, nothing have been processed. I'm a little bit confuesed with it. Anyway, with the old version of this repo, I got the result out keypoints and others, but the old version can only work with the first some frames, I will be Aborted, which may be caused by so many windows with in dokcer.

And, there the latest results. Thanks for your updating and replies.
